lib/rubocop/cop/rails/i18n_locale_texts.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Enforces use of I18n and locale files instead of locale specific strings.
+      #
+      # @example
+      #   # bad
+      #   class Product < ApplicationRecord
+      #     validates :code, format: { with: /\A[a-zA-Z]+\z/, message: "only letters" }
+      #   end
+      #
+      #   # good
+      #   # config/locales/en.yml
+      #   # en:
+      #   #   activerecord:
+      #   #     errors:
+      #   #       models:
+      #   #         product:
+      #   #           code_format: "only letters"
+      #
+      #   class Product < ApplicationRecord
+      #     validates :code, format: { with: /\A[a-zA-Z]+\z/, message: :code_format }
```

Comment:
I couldn't find a mention of this in Rails guides or API. Does `message: :code_format` work?
Do you think it would be possible to come up with a more conventional example like:
```
validates :name, presence: true
```

> The key for the error message in this case is :blank. Active Record will look up this key in the namespaces ... `activerecord.errors.models.[model_name].attributes.[attribute_name]`, ...  it will try the following keys:

```
activerecord.errors.models.user.attributes.name.blank
```

This way, `message: ` doesn't even have to be specified.

Reply:
https://github.com/rails/rails/blob/9b857799eb35c4175027980651d645bee955e148/activemodel/test/cases/validations/i18n_validation_test.rb#L460-L466

Changed to more conventional example to make it simpler and decrease possible confusion.